RestaurantList.js

```diff
@@ -3,33 +3,58 @@ import Restaurant from './Restaurant'
 import RestaurantModel from './RestaurantModel'
 
 class RestaurantList extends React.Component {
-  constructor() {
-    super()
+  constructor(props) {
+    super(props)
     this.displayName = "RestaurantList"
-
     this.state = {
-      restaurants : [
-        new RestaurantModel("Moses", "Burger", 2)
-        ,new RestaurantModel("המזנון", "Pita", 3)
-        ,new RestaurantModel("Fresh", "Bari", 1)
-        ,new RestaurantModel("Hachim", "Mideterian", 3)
-      ]
+      selectedRestaurant : 0
     }
   }
 
+  handleSelectedRestaurant(restaurant, i) {
+    console.log('selected restaurant ' + i)
+    this.setState({ selectedRestaurant: i });
+  }
+
   render() {
+    console.log(this.props.restaurants[this.state.selectedRestaurant].address)
     return (
       <div className="row">
         <div className="col-md-7">
-          <ul class="media-list">
-            {this.state.restaurants.map(function(restaurant, i) {
-              return <Restaurant model={restaurant} key={i} />
-            })}
+          <ul className="media-list">
+            {this.props.restaurants.map(
+              function(restaurant, i) {
+                var selected = (this.state.selectedRestaurant == i)
+                return (<Restaurant
+                  onClick={this.handleSelectedRestaurant.bind(this, restaurant, i)}
+                  model={restaurant}
+                  key={i}
+                  selected={selected} />)
+              }, this)
```

Comment:
I would always run the map outside of the (<el>) and interpolate just the bar name